tests/test_clean.py

```diff
@@ -0,0 +1,111 @@
+import os
+from pathlib import Path
+
+import pytest
+import zstandard as zstd
+
+from fixtures import DataDir, get_mocked_downloads, FIXTURES_PATH
+from pipeline.data import parallel_importer
+
+SRC = "ru"
+TRG = "en"
+CURRENT_FOLDER = os.path.dirname(os.path.abspath(__file__))
+
+
+def add_fake_alignments(corpus):
+    corpus_and_aln = []
+    for line in corpus:
+        parts = line.split("\t")
+        src_sent, trg_sent = parts[0], parts[1]
+        min_len = min(len(src_sent.split()), len(trg_sent.split()))
+        aln = " ".join([f"{idx}-{idx}" for idx in range(min_len)])
+        corpus_and_aln.append(f"{line}\t{aln}")
+
+    return corpus_and_aln
+
+
+# it's very slow to download and run BERT on 2000 lines
+parallel_importer.add_alignments = add_fake_alignments
+
+
+def config(trg_lang, data_dir):
+    if trg_lang == "en":
+        # copy the test config and swap language direction
+        config_path = os.path.abspath(os.path.join(FIXTURES_PATH, "config.pytest.yml"))
+        new_config_path = data_dir.join("config.yml")
+
+        with open(config_path) as f:
+            new_config = f.read().replace(
+                """  src: en
+  trg: ru""",
+                """  src: ru
+  trg: en""",
+            )
+
+        with open(new_config_path, "w") as f:
+            f.write(new_config)
+        return new_config_path
+    elif trg_lang == "zh":
+        return os.path.abspath(os.path.join(FIXTURES_PATH, "config.pytest.enzh.yml"))
+    return None
+
+
+def read_lines(path):
```

Comment:
You can use the `read_lines` utility in pipeline/common/downloads.py.

Reply:
Yes, but it requires using it with `with` an iterator ,which would make the code in the tests a lot more verbose, so I'd keep this simple function for now